dev_guide/builds/build_inputs.adoc

```diff
@@ -0,0 +1,911 @@
+[[dev-guide-build-inputs]]
+= Build Inputs
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[how-build-inputs-work]]
+== How Build Inputs Work
+
+A _build input_ provides source content for builds to operate on. There are
+several ways to provide source in {product-title}. In order of precedence:
+
+* xref:dockerfile-source[Inline Dockerfile definitions]
+* xref:image-source[Content extracted from existing images]
+* xref:source-code[Git repositories]
+* xref:binary-source[Binary inputs]
+* xref:using-secrets-during-build[Input secrets]
+* xref:using-external-artifacts[External artifacts]
+
+Different inputs can be combined into a single build. As the inline Dockerfile takes
+precedence, it can overwrite any other file named *_Dockerfile_* provided by
+another input. Binary input and Git repositories are mutually exclusive inputs.
+
+Input secrets are useful for when you do not want certain resources or
+credentials used during a build to be available in the final application image
+produced by the build. External artifacts outside of a source repository can
+also be used to pull in additional files during a build.
+
+Whenever a build is run:
+
+. A working directory is constructed and all input content is placed in the
+working directory. For example, the input Git repository is cloned into the
+working directory, and files specified from input images are copied into the
+working directory using the target path.
+
+. The build process changes directories into the `contextDir`, if one is
+defined.
+
+. The inline Dockerfile, if any, is written to the current directory.
+
+. The content from the current directory is provided to the build process
+for reference by the Dockerfile, *_assemble_* script, or custom builder logic.
+This means any input content that resides outside the `contextDir` will be
+ignored by the build.
```

Comment:
@bparees Do we need to try adding in mention of build inputs and external artifacts into this numbered list?